tests/splines/matrix.cpp

```diff
@@ -179,6 +179,28 @@ TEST(Matrix, Band)
     solve_and_validate(*matrix);
 }
 
+TEST(Matrix, PDSBand)
+{
+    std::size_t const N = 10;
+    std::size_t const k = 3;
+    std::unique_ptr<ddc::detail::SplinesLinearProblem<Kokkos::DefaultExecutionSpace>> matrix
+            = std::make_unique<
+                    ddc::detail::SplinesLinearProblemPDSBand<Kokkos::DefaultExecutionSpace>>(N, k);
+
+    // Build a positive-definite symmetric full-rank band matrix
+    for (std::size_t i(0); i < N; ++i) {
+        matrix->set_element(i, i, 2.0 * k);
+        for (std::size_t j(std::max(0, int(i) - int(k))); j < i; ++j) {
+            matrix->set_element(i, j, -1.0);
+        }
+        for (std::size_t j(i + 1); j < std::min(N, i + k + 1); ++j) {
+            matrix->set_element(i, j, -1.0);
+        }
+    }
```

Comment:
I am not sure it is actually PDS. I think the matrix we had in mind was supposed to be symmetric and diagonally dominant thus PDS. But it does not seem to be diagonally dominant. Can you check ?

Reply:
This matrix comes from SLL, i will check though
https://gitlab.maisondelasimulation.fr/gysela-developpers/gyselalibxx/-/blob/main/vendor/sll/tests/matrix.cpp?ref_type=heads#L79

Reply:
```
import numpy as np;

N=10;
k=3;

A = np.zeros((10,10));
for i in range(0, N):
  A[i,i] =  2. * k;
  for j in range(max(0,i-k),i):
    A[i, j] = -1.;
  for j in range(i+1, min(N,i+k+1)):
    A[i, j] = -1.;

print(A);
print(np.all(np.linalg.eigvals(A) > 0));
```
![image](https://github.com/CExA-project/ddc/assets/22033714/b36701e1-1ea8-46d3-857a-8e6384a1b4c7)

Reply:
Ok for this (N,k) couple but i would add 1 on the diagonal, this should make it diagonally dominant